### PR TITLE
Autoplay Bug for iOS Users

### LINKED ISSUE #65: No Auto Play Bug
**Is your feature request related to a problem? Please describe.**When the browser doesn't allow auto play the app breaks

**Describe the solution you'd like**
If the browser allows auto play then the music play. If the browser does not then do not play the audio or have a suggestion to play it

### PR DESCRIPTION
**Pull request by BlackCubes**
## Changes
1. Fixed the audio in `PokemonCry` component to play/pause without errors for iOS users.
2. Created an `audioPlayAsync` function that checks if the audio could play depending on the browser and/or operating system.
3. Imported `audioPlayAsync` function in `PokemonCry.js`.
4. Refactored `audioPlay` function in `PokemonCry.js`.

## Purpose
The problem was that the app would throw errors on the browser for iOS users (mainly on Safari), but only when the user is searching for a friendly pokemon. The error would not happen when the user is searching for the enemy. This could be due to a security feature from the Apple company.

## Approach
According to the section "User Control of Downloads Over Cellular Networks" in [Apple's developer documentation](https://developer.apple.com/library/archive/documentation/AudioVideo/Conceptual/Using_HTML5_Audio_Video/Device-SpecificConsiderations/Device-SpecificConsiderations.html), it states that iOS disables autoplay and that play be initiated as part of a user interaction. The reason that the other audios work (e.g. both theme songs and enemy pokemon sound) while the friendly pokemon sound did not is because those audios are being fired immediately after a user interaction such as hitting the submit button.

There are a few potential problems as to why the friendly pokemon sound is not working on with or without the versus screen:
1. There are two many conditional renderings outside and inside the `PokemonCry` component.
2. The conditional rendering on `PokemonsCry` component for the pokeball animation in `SinglePokemon.js` is throwing the error.

For problem 1, there is a conditional rendering inside `PokemonCry.js`, and there is one outside for the pokeball animation in `SinglePokemon.js`. These are probably throwing the error since iOS might be thinking since after a few seconds of waiting to render, the user has not "interacted" anymore. To fix this is to remove both the conditional renderings of `isPokemonCries` and `isPokeballRendering` both inside and outside of `PokemonCry` component, respectively. Because of this, most of the functionalities in `PokemonCry.js` had to be refactored:
- The sound url check for the pokemon sound in the first `useEffect` in `PokemonCry.js` was moved to `App.js` as a function called `getPokeSoundUrl`. This was then used in `getPokeByNameOrIdAPI` in `App.js` in order to let iOS know that this is part of a user interaction.
- The `props` for `PokemonCry.js` was changed from `pokeName` to `pokeSoundUrl` and `isPokeballRendering` (more on `isPokeballRendering` later). This was done in order to fire the first `useEffect` after the user interaction.
- States of `pokeSoundUrl` and `enemyPokeSoundUrl` were created in `App.js` in order to store different urls of the pokemon sound. This helps to ensure that the audio in `PokemonCry` component is not being fired twice by just one state, and it helps to fire off the first `useEffect`.

For problem 2, the problem with conditionally rendering after 1700ms from the pokeball animation is that iOS probably thinks there are no user interaction, and so throws an error. To fix this is simple: Store the `isPokeballRendering` as a prop in `PokemonCry.js`, and add it to the first `useEffect` in `PokemonCry.js` to check in a regular if-conditional on whether the pokeball animation is still playing. This would not block any user interaction for iOS, and thus would no display any errors.

Lastly, an async function of `audioPlayAsync` was created as a backup in order to check if the browser and/or any operating system supports an audio playing immediately. If it does not, then pause the audio. If it does, play the audio.

## Learning
[Apple's developer documentation](https://developer.apple.com/library/archive/documentation/AudioVideo/Conceptual/Using_HTML5_Audio_Video/Device-SpecificConsiderations/Device-SpecificConsiderations.html)

## Screenshots
my-fellow-nerds

Closes #65

### CODE REVIEW

File: pokemon-app/src/App.js [hunk 1]
```diff
@@ -8,21 +8,43 @@ import { getEvolutionChain, getPokeByNameOrId } from "./utils/api";
 import "./App.css";
 import "./styles/Pokedex-model.css";
 import Toad from "./components/toad";
+import pokeCries from "./utils/pokeCries";
+import reactoadSound from "./assets/sounds/reactoad.mp3";
```
Comment (krferrell) [inline pokemon-app/src/App.js:12]: Can't wait for this.

File: pokemon-app/src/App.js [hunk 2]
```diff
@@ -8,21 +8,43 @@ import { getEvolutionChain, getPokeByNameOrId } from "./utils/api";
...
+    let pokeSoundUrlPath = "";
+
+    if (pokeName === "Reactoad" || pokeCries[pokeName]) {
+      pokeSoundUrlPath =
+        pokeName === "Reactoad" ? reactoadSound : pokeCries[pokeName];
+    }
+
+    return pokeSoundUrlPath;
+  };
+
```
Comment (krferrell) [inline pokemon-app/src/App.js:35]: This function is very clean. Nice work.

File: pokemon-app/src/utils/audioPlayAsync.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/**
+ * Checks if the audio could be played depending on the browser and/or operating system.
```
Comment (krferrell) [inline pokemon-app/src/utils/audioPlayAsync.js:2]: This is really cool that it is able to check browser compatibility.

**Review by krferrell — APPROVED** [review]
Nice job!

**Review by kmausolf — COMMENTED** [review]
This is really fascinating!  Great work identifying and solving this problem!

File: pokemon-app/src/components/PokemonCry.js [hunk 4]
```diff
@@ -1,30 +1,37 @@
...
+      // Display the error to the console.
+      console.error(error);
+    }
+  };
+
+  useEffect(() => {
+    // If sound url exists and the pokeball animation is not rendering,
+    // continue.
+    if (pokeSoundUrl.length && !isPokeballRendering) {
+      // Set the cries to true in order to trigger the second useEffect.
```
Comment (Gjhnsn) [inline pokemon-app/src/components/PokemonCry.js:31]: very interesting, great comments through your code

**Review by Gjhnsn — APPROVED** [review]
Really cool you were able to check the browser/OS

**Review by Wolftownn — APPROVED** [review]
Great job Elias! I really appreciate the thoroughness in your PR description as well. A master!

**Pull request merged**